### ISSUE TITLE
FSSA Patch 1.0.8.0 Light Isuues

### ISSUE THREAD

**Issue by TP555** [elishacloud/BloodRayne-2-Patch#7]
Hi

I have some Light Issues , when the DX9 Wrapper is Active , without or with DGVoodoo 2.71.3 , no Light Pops

I did Try , all your Previous Versions , and the D3D8to9 from Crossire , with every the same Result.
Reshade 4.8.2 couldn't fix this issue
And of course with and without HD Textures. !

Combine DGVoodoo with your d3d8patch.dll , which load all hacks HD Textures , was not succesfully

https://www.youtube.com/watch?v=26fmNR3pGSk

Try to fix this , or the best way i think is DX11 with DGVoodoo + your Fix with better Graphic Settings  HD Textures. etc.

**Comment by elishacloud** [comment]
> the best way i think is DX11 with DGVoodoo + your Fix with better Graphic Settings HD Textures. etc.

You can do this.  Just use the d3d9.dll from dgVoodoo2 to upgrade from DirectX9 to DirectX11.  Or you can use [dxvk](https://github.com/doitsujin/dxvk) to update from DirectX9 to Vulkan.

**Comment by TP555** [comment]
Re

I will try this , and with DXVK !

**Comment by romulus2k4** [comment]
Hi.
The lighting issue you've reported is actually how the game itself behaves and isn't a fault of anything else but the game's engine.

**Comment by TP555** [comment]
Re

@romulus2k4 

Realy No issues , in Original Game and with DGVoodoo , is Just with the DX9 Wrapper

With Pure DX8 Mode ! , to see isn't Engine or Game related , or is an General Original Game issue   !

https://www.youtube.com/watch?v=HLkfoXz__3U

@elishacloud

with d3d9.dll from DGVoodoo , it dosn't work anymore (Crash)
Older Version 2.62.3 , did run , but the HD Textures and another , didn't work anymore (d3d8patch.dll)

Temporary Fix . rename d3d8patch.dll to d3d8.dll , for pure DX8 Mode  !
But OBS Studio , breaks the FPS from 110 to 57 or less , so i must down to 30FPS for Recording , and MSI AfterBurner won't show up in the Recording anymore !
And Texture Caching . during the Gameplay , brakes the Performance much more , and Textures will be slow fully loaded  !

An Precache for the Textures , will be nice  !

**Comment by elishacloud** [comment]
> Realy No issues , in Original Game and with DGVoodoo , is Just with the DX9 Wrapper

The issue here may be with the video card drivers.  It is even possible that you have a different video card driver for DirectX8 than you do for DirectX9, thus the difference.  dgVoodoo2 actually emulates DirectX so it is not directly interpreted by the driver, it adds a layer between the game and your video card driver.  However, d3d8to9 simply converts DirectX8 directly to DirectX9.  There is no emulation happening here.

> with d3d9.dll from DGVoodoo , it dosn't work anymore (Crash)

This looks like a bug with dgVoodoo2.  It should not be crashing.

I will create an update to make d3d8to9 optional and allow the FSSA patch to work with other DirectX8 wrappers, like dgVoodoo2.  I will also add an option for fullscreen windowed mode.

**Comment by elishacloud** [comment]
From what I can tell the lighting issues are caused by d3d8to9.  I suspect it is an issue with one of the shaders.

However, I made some new options to allow you to enable fullscreen windowed mode and allow you to disable d3d8to9, which seems to fix the lighting issue.

Here is the update:  [patch.zip](https://github.com/elishacloud/BloodRayne-2-Patch/files/5492422/patch.zip)

_Edit: you can modify the `d3d8.ini` file to change the settings._

**Comment by TP555** [comment]
Re

Cool , this works awesome now  :+1: 

now DG Voodoo 2.71.3 works too , and Reshade 4.8.2  !!

@all

how to

Edit the d3d8.ini

```ini
; Path to a d3d8.dll file to use as a DirectX8 wrapper.  Requires 'd3d8to9 = 0'.
D3d8WrapperPath = d3d8v.dll

; Causes BloodRayne 2 to use DirectX 9. (0|1)
d3d8to9 = 0

; Enables the FSSA Patch BloodRayne 2 'BR2 FSAA Patch 1.666'. (0|1)
EnableFSSAPatch = 1

; Enables fullscreen windowed Mode.  (0|1)
FullscreenWndMode = 0
```

then extract d3d8.dll from dgvoodoo 2.71.3 and rename it how you want , and edit it in the d3d8.ini
in my case d3d8v.dll
then run dgVoodooCpl.exe , and configure all , importment is set the Video Ram to 2048 in the DirectX TAB.

BTB

now im looking, if i can , improve the Texture Caching , i did seen , the Zip Files are 50% or more packed , i think this needs a momment to extract it in the VRAM , and this cause the Texture Poping at begin , and this needs about 60% CPU Usage , and this cause some stuttering , when it done , it goes down to 30% or less  !

i will try to repack all 4 Archives but 100% without Compression  !!

But mayby can you look , if you can improve , or add a full Texture Cache , during Loading Scrren  !!

Edit , ok is much better without Compresssion , not 100% perfect , but ok , and i think less a problem , for faster Computers
i have an i5 2400 @3,00ghz

But a full Texture Precache will be nice   !

**Comment by romulus2k4** [comment]
But not using the d3d8to9 disables the extra visual effects that FSAA Patch provides, correct?

Anyways, in about 12 days, BloodRayne Terminal Cut is about to drop.

**Comment by elishacloud** [comment]
> But not using the d3d8to9 disables the extra visual effects that FSAA Patch provides, correct?

No, it should not.  All of the FSAA Patch should be available whether you have `d3d8to9` enabled or not.

**Comment by romulus2k4** [comment]
> > But not using the d3d8to9 disables the extra visual effects that FSAA Patch provides, correct?
> 
> No, it should not. All of the FSAA Patch should be available whether you have `d3d8to9` enabled or not.

Hello Elisha. After doing some testing, I can confirm the following findings:

Disabling the d3d8to9 that ships with the FSAA patch does disable each and every FSAA patch based effects. Replacing the included d3d8to9 with dgVoodoo2 or crosire's own wrapper does the same thing.

You can confirm this by simply having AA enabled for dynamic shadows for the reflections which doesn't get applied or have a look at the in game blood.


> Re
> 
> @romulus2k4
> 
> Realy No issues , in Original Game and with DGVoodoo , is Just with the DX9 Wrapper
> 
> With Pure DX8 Mode ! , to see isn't Engine or Game related , or is an General Original Game issue !
> 
> https://www.youtube.com/watch?v=HLkfoXz__3U
> 

Hi. The lighting issue is introduced by the FSAA patch itself. If you use the D3D8 code path or dgVoodoo2 or some other wrapper, the FSAA patch doesn't get applied, therefore resulting in the vanilla behavior aka the lighting not flickering.

**Comment by elishacloud** [comment]
Hello @romulus2k4, thanks for the testing.

> Disabling the d3d8to9 that ships with the FSAA patch does disable each and every FSAA patch based effects.

Should be fixed now.  Disabling the d3d8to9 that ships with the FSAA patch should keep the FSSA patch effects in place.

> The lighting issue is introduced by the FSAA patch itself.

This does not appear to be the case.  The lighting issue seems to be caused by d3d8to9.  I will look into it later.

> If you use the D3D8 code path or dgVoodoo2 or some other wrapper, the FSAA patch doesn't get applied

Fixed.  However dgVoodoo2 does not work with the FSAA patch.  There seems to be some bug in dgVoodoo2.

Here is the new update: [patch.zip](https://github.com/elishacloud/BloodRayne-2-Patch/files/5580615/patch.zip)

**Comment by elishacloud** [comment]
Closing issue.
